client/InvoiceTickets.js

```diff
@@ -10,10 +22,38 @@ Template.InvoiceTickets.helpers({
     return InvoiceTickets.byTimeRange(
       FlowRouter.getParam("filter"),
       FlowRouter.getQueryParam("sortBy"),
-      FlowRouter.getQueryParam("sortOrder")
+      FlowRouter.getQueryParam("sortOrder"),
+      Session.get("itemsLimit")
     );
   },
   formatDate: function(date){
     return moment(date).format("MM-DD-YYYY");
   },
 });
+Template.InvoiceTickets.moreResults = function() {
+    // If, once the subscription is ready, we have less rows than we
+    // asked for, we've got all the rows in the collection.
+    return !(InvoiceTickets.find().count() < Session.get("itemsLimit"));
+}
+
+// whenever #showMoreResults becomes visible, retrieve more results
+function showMoreVisible() {
+    var threshold, target = $("#showMoreResults");
+    if (!target.length) return;
+
+    threshold = $(window).scrollTop() + $(window).height() - target.height();
+    if (target.offset().top <= threshold) {
+        if (!target.data("visible")) {
+            target.data("visible", true);
+            Session.set("itemsLimit",
+                Session.get("itemsLimit") + ITEMS_INCREMENT);
+        }
+    } else {
+        if (target.data("visible")) {
+            target.data("visible", false);
```

Comment:
There is a more elegant way than using a `visible` data variable to solve the problem of too many calls of the `showMoreVisible` function. Checkout `_.throttle` in the underscore library and think about how this can help your here! :wink: